bundle/edu.gemini.qpt.shared/src/main/java/edu/gemini/qpt/shared/util/EnumPio.java

```diff
@@ -25,6 +29,37 @@ public static ParamSet getEnumParamSet(final PioFactory factory, final String na
         return params;
     }
 
+    private static final Map<String, String> CLASS_NAME_UPDATES;
+
+    static {
+        final Map<String, String> m = new HashMap<>();
+        m.put(
+            "edu.gemini.spModel.gemini.gmos.GmosNorthType$DetectorManufacturerNorth",
+            "edu.gemini.spModel.gemini.gmos.GmosCommonType$DetectorManufacturer"
+        );
+        m.put(
+            "edu.gemini.spModel.gemini.gmos.GmosSouthType$DetectorManufacturerSouth",
+            "edu.gemini.spModel.gemini.gmos.GmosCommonType$DetectorManufacturer"
+        );
+        m.put(
+            "edu.gemini.qpt.core.sp.Inst",
+            Inst.class.getName()
+        );
+        m.put(
+            "edu.gemini.spModel.gemini.gems.Canopus$Wfs$1",
+            CanopusWfs.cwfs1.getClass().getName()
+        );
+        m.put(
+            "edu.gemini.spModel.gemini.gems.Canopus$Wfs$2",
+            CanopusWfs.cwfs2.getClass().getName()
+        );
+        m.put(
+            "edu.gemini.spModel.gemini.gems.Canopus$Wfs$3",
+            CanopusWfs.cwfs3.getClass().getName()
+        );
+        CLASS_NAME_UPDATES = Collections.unmodifiableMap(m);
+    }
```

Comment:
I'm not proud of this, but it is in keeping with similar changes in the past.  The correct way to do this would be in `Schedule.java` but would require some refactoring.  The `facilities` `MixedEnumSet` is read before the migrations happen in the `Schedule` constructor.  Instead we need to apply migrations like these before reading the `facilities`.  In lieu of that, I've just done the same workaround we did in the past.

Reply:
Yep, looks fine to me.